connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,99 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Test;
+
+import java.io.*;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConnectionTest {
+    private static Connection connection;
+    private static ByteArrayOutputStream innerOutputStream;
+    private static int originalNumberOfBytes;
+    private static final SocketAddress address = mock(SocketAddress.class);
+    private static final Message messageToReceive = new Message(TEXT, "MessageToReceive");
+    private static final Message messageToSend = new Message(TEXT, "MessageToSend");
+    private static final Message testMessage = new Message(TEXT, "InOrderNotToGetEOFException");
+
+    private void setUpConnectionObjectWithByteStreams() throws Exception {
+        ByteArrayOutputStream outerOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream outerOOS = new ObjectOutputStream(outerOutputStream)) {
+            outerOOS.writeObject(messageToReceive);
+        }
+
+        innerOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream oos = new ObjectOutputStream(innerOutputStream)) {
+            oos.writeObject(testMessage);
+        }
+        originalNumberOfBytes = innerOutputStream.toByteArray().length;
+
+        ByteArrayInputStream innerInputStream = new ByteArrayInputStream(outerOutputStream.toByteArray());
```

Comment:
Firstly you said:
:apple: Issue
You should initialize variable at the closest point possible to its intended usage.
I placed it more closer to its intended usage.
Now you say that I should place it above as it was in the previous commit.
I really don't understand what you want.